test/test_valgrind.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+# Copyright (C) 2024 Intel Corporation
+# Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+set -e
+
+if ! valgrind --version > /dev/null; then
+	echo "error: valgrind not found"
+	exit 1
+fi
+
+if [ "$1" == "" ]; then
+	echo "Usage: $(basename $0) <memcheck|drd|helgrind>"
+	exit 1
+fi
+
+case $1 in
+memcheck)
+	OPTION="--leak-check=full"
+	;;
+drd)
+	OPTION="--tool=drd"
+	;;
+helgrind)
+	OPTION="--tool=helgrind"
+	;;
+*)
+	echo "error: unknown option: $1"
+	exit 1
+	;;
+esac
+
+FAIL=0
+
+echo "Running: \"valgrind $OPTION\" for the following tests:"
+
+for tf in $(ls -1 ./test/umf_test-*); do
```

Comment:
hmm.. I can see you're relying on being in the build dir - perhaps you should (at least) mention this in the script description.

You can also check if the number of tests are `> 0` - if not this script should fail, I guess

Reply:
Done